server/clcachesrv.py

```diff
@@ -114,9 +127,24 @@ def onSigterm(handle, signum):
 def main():
     logging.basicConfig(format='%(asctime)s [%(levelname)s]: %(message)s', level=logging.INFO)
 
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('--exclude', metavar='REGEX', action='append', \
+                        help='Regex ( re.search() ) for exluding of directory watching. Can be specified multiple times. \
+                              Pathnames to test against are after os.path.normcase(), i.e. all lowercase with backslashes. \
+                              Example: --exclude \\\\build\\\\')
+    parser.add_argument('--disable_watching', action='store_true', help='Disable watching of directories which we have in the cache.')
+    args = parser.parse_args()
+
+    if args.exclude:
+        for pattern in vars(args)['exclude']:
```

Comment:
Not sure I read this correctly, but why do you need to use `vars` here? It seems to me that `exclude` should be a list in itself already:

```python
>>> import argparse
>>> p = argparse.ArgumentParser()
>>> p.add_argument('--exclude', action='append')
_AppendAction(option_strings=['--exclude'], dest='exclude', nargs=None, const=None, default=None, type=None, choices=None, help=None, metavar=None)
>>> args = p.parse_args('--exclude this --exclude that --exclude something_else'.split())
>>> type(args.exclude)
<class 'list'>
>>> args.exclude
['this', 'that', 'something_else']
```

...so I'd expect that one could just use

```python
for pattern in args.exclude or []:
    logging.info("Not watching paths which match: %s", pattern)
```